govuk_chat_evaluation/rag_answers/evaluate.py

```diff
@@ -1,31 +1,225 @@
+):
+    """
+    Function to run the evaluation, aggregate the results, and export them to files.
 
+    Args:
+        _output_dir: The directory to save the evaluation results.
+        evaluation_data_path: Path to the JSONL file containing the evaluation data.
+        evaluation_config: Configuration for the evaluation.
+        deepeval_evaluate_params: Parameters to be passed to deepeval evaluation.
+    """
+    # set DeepEval results folder
+    os.environ["DEEPEVAL_RESULTS_FOLDER"] = str(_output_dir)
 
-def run_deepeval_evaluation(models: list[EvaluationTestCase]):
-    llm_test_cases = [model.to_llm_test_case() for model in models]
+    # convert raw data into model instances for evaluation
+    models = jsonl_to_models(evaluation_data_path, EvaluationTestCase)
 
-    # contrived example metric
-    correctness_metric = GEval(
-        name="Correctness",
-        criteria="Determine if the 'actual output' is correct based on the 'expected output'.",
-        evaluation_params=[
-            LLMTestCaseParams.ACTUAL_OUTPUT,
-            LLMTestCaseParams.EXPECTED_OUTPUT,
-        ],
-        threshold=0.5,
+    # initialise EvaluationResults with the config and the cases
+    evaluation_results = EvaluationResults(
+        evaluation_config=evaluation_config,
+        cases=[model.to_llm_test_case() for model in models]
     )
 
-    deepeval_evaluate(llm_test_cases, [correctness_metric], print_results=False)  # type: ignore
+    # run the DeepEval evaluation
+    evaluation_results.run_deepeval_evaluation(**deepeval_evaluate_params)
+
+    # initialise AggregatedResults to aggregate the results
+    aggregation = AggregatedResults(evaluation_results.evaluation_results)
+
+    # calculate aggregated results and exports results to CSV files
+    aggregation.export_to_csvs(_output_dir, prefix="alessia_test")
+
+    print("Evaluation Results:")
+    print(aggregation.summary)
+
+
+class EvaluationResults:
+    def __init__(self, evaluation_config: EvaluationConfig, cases: list[LLMTestCase]):
+        self.evaluation_config = evaluation_config
+        self.metrics = evaluation_config.get_metric_instances()
+        self.llm_judge = evaluation_config.llm_judge_instance
+        self.n_runs = evaluation_config.n_runs
+        self.cases = cases
+        self.evaluation_results : list[TestResult] = []
+        
+    def run_deepeval_evaluation(self, **kwargs) -> None:
+        """"
+        Run the Deepval evaluation on the given models and metrics
+
+        Args:
+            cases : List of test cases to evaluate
+            metrics : List of metrics to use for evaluation
+            n_runs : Number of runs to perform for the evaluation
+            **kwargs: Additional arguments to pass to the deepeval.evaluation function
+
+        Returns:
+            List of evaluation results for each model
+
+        """
+
+        with Timer("Running Deepval Evaluation"):
+            logger.info("Running Deepval evaluation")
```

Comment:
```suggestion
        with Timer("Running Deepeval Evaluation"):
            logger.info("Running Deepeval evaluation")
```

Couple of minor typos